pkg/log/locallogger/locallogger.go

```diff
@@ -40,6 +43,25 @@ func (ll localLogger) Log(keyvals ...interface{}) error {
 	return ll.logger.Log(keyvals...)
 }
 
+func CleanUpRenamedDebugLogs(cleanupPath string, logger log.Logger) {
+	// We renamed the debug log file from debug.log to debug.json for compatibility with support tools.
+	// Check to see if we have any of the old debug.log files still hanging around, and clean them up
+	// if so. The current one is always named debug.log, and rotated files are in the format
+	// `debug-<date>.log.gz`. We do not return an error if we can't clean up these files -- it's not
+	// a big deal.
+	legacyDebugLogPattern := filepath.Join(cleanupPath, "debug*.log*")
+	filesToCleanUp, err := filepath.Glob(legacyDebugLogPattern)
+	if err != nil {
+		level.Error(logger).Log("msg", "could not glob for legacy debug log files to clean up", "pattern", legacyDebugLogPattern, "err", err)
+	} else {
+		for _, fileToCleanUp := range filesToCleanUp {
+			if err := os.Remove(fileToCleanUp); err != nil {
+				level.Error(logger).Log("msg", "could not clean up legacy debug log file", "file", fileToCleanUp, "err", err)
+			}
+		}
+	}
+}
```

Comment:
sorry to jump in an nit pick ... but we can just return if err != nil and reduce nesting

```suggestion
	if err != nil {
		level.Error(logger).Log("msg", "could not glob for legacy debug log files to clean up", "pattern", legacyDebugLogPattern, "err", err)
		return
	}
        for _, fileToCleanUp := range filesToCleanUp {
	        if err := os.Remove(fileToCleanUp); err != nil {
		        level.Error(logger).Log("msg", "could not clean up legacy debug log file", "file", fileToCleanUp, "err", err)
	        }
        }
}
```

Reply:
Updated, thanks!